File: ios/Listen/DesignSystem/Icon.swift
//
//  Icon.swift
//  listen — maps the Lucide icon names used in the design spec to SF Symbols.
//

import SwiftUI

enum IconName: String {
    case audioLines = "audio-lines"
    case menu, plus, x
    case chevronRight = "chevron-right"
    case chevronLeft = "chevron-left"
    case penLine = "pen-line"
    case fileUp = "file-up"
    case fileText = "file-text"
    case layers
    case search
    case settings
    case logOut = "log-out"
    case layoutDashboard = "layout-dashboard"
    case library
    case disc3 = "disc-3"
    case check
    case copy
    case save
    case trash = "trash-2"
    case pencil
    case download
    case focus
    case type
    case unfoldVertical = "unfold-vertical"
    case unfoldHorizontal = "unfold-horizontal"
    case alertTriangle = "alert-triangle"
    case info
    case clock
    case play
    case pause
    case stop = "square"
    case skipBack = "skip-back"
    case skipForward = "skip-forward"
    case heart
    case mail
    case user

    var symbolName: String {
        switch self {
        case .audioLines: return "waveform"
        case .menu: return "line.3.horizontal"
        case .plus: return "plus"
        case .x: return "xmark"
        case .chevronRight: return "chevron.right"
        case .chevronLeft: return "chevron.left"
        case .penLine: return "pencil.line"
        case .fileUp: return "arrow.up.doc"
        case .fileText: return "doc.text"
        case .layers: return "square.3.layers.3d"
        case .search: return "magnifyingglass"
        case .settings: return "gearshape"
        case .logOut: return "rectangle.portrait.and.arrow.right"
        case .layoutDashboard: return "square.grid.2x2"
        case .library: return "books.vertical"
        case .disc3: return "opticaldiscdrive"
        case .check: return "checkmark"
        case .copy: return "doc.on.doc"
        case .save: return "square.and.arrow.down"
        case .trash: return "trash"
        case .pencil: return "pencil"
        case .download: return "arrow.down.circle"
        case .focus: return "viewfinder"
        case .type: return "textformat.size"
        case .unfoldVertical: return "arrow.up.and.down"
        case .unfoldHorizontal: return "arrow.left.and.right"
        case .alertTriangle: return "exclamationmark.triangle"
        case .info: return "info.circle"
        case .clock: return "clock"
        case .play: return "play.fill"
        case .pause: return "pause.fill"
        case .stop: return "stop.fill"
        case .skipBack: return "backward.end.fill"
        case .skipForward: return "forward.end.fill"
        case .heart: return "heart"
        case .mail: return "envelope"
        case .user: return "person"
        }
    }
}

struct Icon: View {
    let name: IconName
    var size: CGFloat = 20
    var color: Color = Theme.fg2
    var filled: Bool = false

    var body: some View {
        Image(systemName: filled && name == .heart ? "heart.fill" : name.symbolName)
            .font(.system(size: size, weight: .regular))
            .foregroundStyle(color)
    }
}

File: android/app/src/main/java/com/listen/app/designsystem/Icon.kt
package com.listen.app.designsystem

/**
 * listen — maps the Lucide icon names used in the design spec to Material
 * Icons (port of ios/Listen/DesignSystem/Icon.swift, which does the same
 * mapping to SF Symbols).
 */

import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material.icons.outlined.FavoriteBorder
import androidx.compose.material3.Icon as Material3Icon
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

enum class IconName {
    AudioLines, Menu, Plus, X,
    ChevronRight, ChevronLeft,
    PenLine, FileUp, FileText, Layers, Search, Settings, LogOut,
    LayoutDashboard, Library, Disc3, Check, Copy, Save, Trash, Pencil,
    Download, Focus, Type, UnfoldVertical, UnfoldHorizontal,
    AlertTriangle, Info, ClockIcon, Play, Pause, Stop, SkipBack, SkipForward,
    Heart, Mail, User, Eye, EyeOff,
}

private fun vectorFor(name: IconName, filled: Boolean): ImageVector = when (name) {
    IconName.AudioLines -> Icons.Default.GraphicEq
    IconName.Menu -> Icons.Default.Menu
    IconName.Plus -> Icons.Default.Add
    IconName.X -> Icons.Default.Close
    IconName.ChevronRight -> Icons.Default.ChevronRight
    IconName.ChevronLeft -> Icons.Default.ChevronLeft
    IconName.PenLine -> Icons.Default.Edit
    IconName.FileUp -> Icons.Default.UploadFile
    IconName.FileText -> Icons.Default.Description
    IconName.Layers -> Icons.Default.Layers
    IconName.Search -> Icons.Default.Search
    IconName.Settings -> Icons.Default.Settings
    IconName.LogOut -> Icons.Default.Logout
    IconName.LayoutDashboard -> Icons.Default.GridView
    IconName.Library -> Icons.Default.LibraryBooks
    IconName.Disc3 -> Icons.Default.Album
    IconName.Check -> Icons.Default.Check
    IconName.Copy -> Icons.Default.ContentCopy
    IconName.Save -> Icons.Default.Save
    IconName.Trash -> Icons.Default.Delete
    IconName.Pencil -> Icons.Default.Edit
    IconName.Download -> Icons.Default.Download
    IconName.Focus -> Icons.Default.CenterFocusStrong
    IconName.Type -> Icons.Default.FormatSize
    IconName.UnfoldVertical -> Icons.Default.UnfoldMore
    IconName.UnfoldHorizontal -> Icons.Default.SwapHoriz
    IconName.AlertTriangle -> Icons.Default.WarningAmber
    IconName.Info -> Icons.Default.Info
    IconName.ClockIcon -> Icons.Default.Schedule
    IconName.Play -> Icons.Default.PlayArrow
    IconName.Pause -> Icons.Default.Pause
    IconName.Stop -> Icons.Default.Stop
    IconName.SkipBack -> Icons.Default.SkipPrevious
    IconName.SkipForward -> Icons.Default.SkipNext
    IconName.Heart -> if (filled) Icons.Default.Favorite else Icons.Outlined.FavoriteBorder
    IconName.Mail -> Icons.Default.Email
    IconName.User -> Icons.Default.Person
    IconName.Eye -> Icons.Default.Visibility
    IconName.EyeOff -> Icons.Default.VisibilityOff
}

@Composable
fun AppIcon(
    name: IconName,
    size: Dp = 20.dp,
    color: Color = Theme.fg2,
    filled: Boolean = false,
    modifier: Modifier = Modifier,
) {
    Material3Icon(
        imageVector = vectorFor(name, filled),
        contentDescription = null,
        tint = color,
        modifier = modifier.size(size),
    )
}
